### ISSUE TITLE
Taking a photo:  Error - Specified cast is not allowed.

### ISSUE THREAD

**Issue by Jugs0101** [jamesmontemagno/MediaPlugin#516]
Hi, go an issue with taking a photo. Can't for the life of me make it work.

## Issue Information
Trying to take a picture with the camera using the example code:

 var file = await CrossMedia.Current.TakePhotoAsync(new Plugin.Media.Abstractions.StoreCameraMediaOptions
            {
                Directory = "Sample",
                Name = "test.jpg"
            });
I get an exception thrown: Specific cast is not allowed.

I do the normal 
 if (!CrossMedia.Current.IsCameraAvailable || !CrossMedia.Current.IsTakePhotoSupported)
which the code gets past no problem.

I've tried a variety of media options but just keep getting the same error.

Anyone help? I'm not sure if it's even to do with the MediaPlugin or some other configuration thing, as I can't believe I'm the only one having this problem.
Any pointers would be most appreciated.

Cheers,

Jugs

======================================

Version Number of Plugin: V3.1.3
Device Tested On: Motorola MotoG 2014, BlackView 7
Simulator Tested On: None
Version of VS: 2017
Version of Xamarin: 2.5.0.280555
Versions of other things you are using: Windows 10 16299.309

**Comment by prashantvc** [comment]
@Jugs0101 that's rather strange! I am not able to reproduce this issue.
Could you try clone the source code and try debugging?

**Comment by Jugs0101** [comment]
Hi Prashant,

Thanks for responding.

I have tried setting up the project from scratch a number of times, and yes, debugging it throws the exception at that line. I have also reinstalled VS 2017. It's also a brand new computer with virtually nothing else on it.

Would more detail on the actual exception be of help? It talks about Java.Interop.Tools.TypeNameMappings.JniType.ToJniNameFromAttributes and Java.Interop.Tools.TypeNameMappings.JniType.ToJniName, but it's such a big load of text it's difficult to decipher what it thinks the problem is.

I'm also new to Xamarin development, but not development in general.

Any further help would be most appreciated.

J
